### ISSUE TITLE
redo-enabled changefeed get stuck

### ISSUE THREAD

**Issue by AkiraXie** [pingcap/tiflow#10462]
### What did you do?

1. open tidb sink changefeed (with  redo and syncpoint enabled )  and s3 sink changefeed
2. tikv resolve ts cannot advance because of https://github.com/tikv/tikv/issues/16340
3. then tikv upgraded and resolve ts advanced
![image](https://github.com/pingcap/tiflow/assets/50363732/9e880d84-39bf-41e0-af4c-6a6ac29eb973)

### What did you see instead?

1. when tikv resolve ts got stuck , changefeeds' checkpoint couldn't advance
2. when tikv resolve ts advanced, s3 sink changefeed checkpoint advanced but redo tidb sink changefeed get stuck yet and eventually in error.
![image](https://github.com/pingcap/tiflow/assets/50363732/568cf187-39dc-42a2-a615-7471f0ee6d67)
![img_v3_0271_af5f294b-92a4-4746-979d-874551e0a57g](https://github.com/pingcap/tiflow/assets/50363732/085ab1e1-d1d4-4172-ad27-0c1a9c992fee)

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
sh-5.1# /cdc version
Release Version: v7.6.0-alpha
Git Commit Hash: bfc74c3abf62904481ad31b4c794de1ae43c6a1b
Git Branch: heads/refs/tags/v7.6.0-alpha
UTC Build Time: 2024-01-10 11:37:33
Go Version: go version go1.21.5 linux/amd64
Failpoint Build: false
```

**Comment by CharlesCheung96** [comment]
1. From the log of owner, we can find that the resolvedTs of table-210261 is stuck:
![image](https://github.com/pingcap/tiflow/assets/61726649/c13e72f1-eda3-4b0b-897b-77c2605f7a27)

2. From the log of ticdc-0, we can find that the kv client of table-210261 is stuck for up to 24 hours:
![image](https://github.com/pingcap/tiflow/assets/61726649/9215bac0-f895-4b5e-8454-30557b32546e)
![image](https://github.com/pingcap/tiflow/assets/61726649/211a37ca-d4cd-4e33-8269-7b11cb9eca02)

3. Form the log of tikv-12, we can find that the leader of region 13466505 is stuck for up to 24 hours:
![img_v3_0275_f25de697-9b4f-4fba-947c-516e72a807fg](https://github.com/pingcap/tiflow/assets/61726649/65612354-73f0-48a1-8383-9a17b71cb7c7)

In conclusion, this issue was caused by the tikv issue [#16340](https://github.com/tikv/tikv/issues/16340). Besides, we also find that [the monitor of max gap of resolvedTs is inaccurate](https://github.com/tikv/tikv/issues/16390).

**Comment by CharlesCheung96** [comment]
For ticdc, the ticdc_scheduler_slow_table_puller_resolved_ts_lag is also inaccurate: It calculates the progress of the table with the slowest checkpoint. However, what we really care about is the puller progress of the table with the smallest resolvedTs.
